scheduler/src/cook/components.clj

```diff
@@ -291,6 +292,14 @@
                                   offer-incubate-ms)
      :mesos-master (fnk [[:config [:mesos master]]]
                         master)
+     :mesos-master-hosts (fnk [[:config [:mesos master {master-hosts nil}]]]
+                              (if master-hosts
+                                (if (and (sequential? master-hosts) (every? string? master-hosts))
+                                  master-hosts
+                                  (throw (ex-info ":mesos-master should be a list of hostnames (e.g. [\"host1.example.com\", ...])" {})))
+                                (->> master
+                                     (re-seq #"[/|,]?([^/,:]+):\d+")
+                                     (map second))))
```

Comment:
Can you make this a `mapv` or follow it with a `doall` so that regex parse failures trigger sooner? Also, validate that the result of this passes `(every? string? %)`